src/Microsoft.ML/CSharpApi.cs

```diff
@@ -6173,7 +6161,7 @@ public enum KMeansPlusPlusTrainerInitAlgorithm
         /// <summary>
         /// K-means is a popular clustering algorithm. With K-means, the data is clustered into a specified number of clusters in order to minimize the within-cluster sum of squares. K-means++ improves upon K-means by using a better method for choosing the initial cluster centers.
         /// </summary>
-        public sealed partial class KMeansPlusPlusClusterer : Microsoft.ML.Runtime.EntryPoints.CommonInputs.ITrainerInput, Microsoft.ML.ILearningPipelineItem
+        public sealed partial class KMeansPlusPlusClusterer : Microsoft.ML.Runtime.EntryPoints.CommonInputs.IUnsupervisedTrainerWithWeight, Microsoft.ML.Runtime.EntryPoints.CommonInputs.ITrainerInput, Microsoft.ML.ILearningPipelineItem
```

Comment:
Is this intentionally included in this PR?

Reply:
yes and no.
We don't run GenerateCsharpApi test, so sometimes people forget to update this file. It's a quite old PR, so I had this change, but now I merge it with master, and change is gone, since someone else update CSharpApi file.

---
In reply to: [193298476](https://github.com/dotnet/machinelearning/pull/286#discussion_r193298476) [](ancestors = 193298476)